status-im/status-react
# Fix failing ERC-1155 collectible transactions

fixes #20973 

### Summary

For some reason we were parsing the `token-id` into an int, which caused a significant loss of precision (see below) and we ended up with a different `token-id` altogether. As a result, the collectible was not found in the contract balance and the tx failed with  `ERC1155: insufficient balance for transfer` or `can't substract more than there is`. 

Most likely non-ERC1155 transactions were not failing due to smaller `token-id`s, which didn't lead to the loss of precision from `js/parseInt`.

```cljs
(-> "49303604037254424499205758826317261461258103209481173801878855324184340332571"
    js/parseInt ;; => 4.9303604037254426E76
    utils.money/to-hex
    ;; Converting token-id back
    utils.money/bignumber
    utils.money/to-string)
;; => "49303604037254426018823311161424344263693657511051997943531504659167293997056"
```

### Review notes


### Testing notes


#### Platforms


- Android
- iOS

#### Areas that maybe impacted


##### Functional
- wallet / transactions

### Steps to test


- Open Status
- Go to collectibles
- Open an ERC-1155 collectible (e.g. any from [here](https://rarible.com/collection/0xb66a603f4cfe17e3d27b87a8bfcad319856518b8/items))
- Send collectible
- Check etherscan if tx succeeded

status: ready
